firebase_admin/mlkit.py

```diff
@@ -459,6 +550,85 @@ def __init__(self, app):
         self._client = _http_client.JsonHttpClient(
             credential=app.credential.get_credential(),
             base_url=self._project_url)
+        self._operation_client = _http_client.JsonHttpClient(
+            credential=app.credential.get_credential(),
+            base_url=_MLKitService.OPERATION_URL)
+
+    def get_operation(self, op_name):
+        _validate_and_parse_operation_name(op_name)
+        try:
+            return self._operation_client.body('get', url=op_name)
+        except requests.exceptions.RequestException as error:
+            raise _utils.handle_platform_error_from_requests(error)
+
+    def handle_operation(self, operation, max_polling_attempts=None, max_time_seconds=None,
+                         always_return_model=False):
+        """Handles long running operations.
+
+        Args:
+            operation: The operation to handle.
+            max_polling_attempts: The maximum number of polling requests to make.
+                (None for no limit)
+            max_time_seconds: The maximum seconds to try polling for operation complete.
+                (None for no limit)
+            always_return_model: If true, returns a locked Model instead of raising deadline
+                exceeded exceptions.
+
+        Returns:
+            dict: A dictionary of the returned model properties.
+
+        Raises:
+            TypeError: if the operation is not a dictionary.
+            ValueError: If the operation is malformed.
+        """
+        if not isinstance(operation, dict):
+            raise TypeError('Operation must be a dictionary.')
+        op_name = operation.get('name')
+        _, model_id = _validate_and_parse_operation_name(op_name)
+
+        current_attempt = 0
+        start_time = datetime.datetime.now()
+        stop_time = (None if max_time_seconds is None else
+                     start_time + datetime.timedelta(seconds=max_time_seconds))
+        while True:
```

Comment:
`while not operation.get('done'):` might simplify this loop a bit.

Reply:
done